tests/test_mnemonic_generation.py

```diff
@@ -42,6 +43,31 @@ def test_calculate_checksum():
     assert bip39.mnemonic_is_valid(" ".join(mnemonic))
 
 
+def test_calculate_checksum_invalid_mnemonics():
+    """
+        Should raise an Exception on a mnemonic that is invalid due to length or using invalid words.
+    """
+    with pytest.raises(Exception) as e:
+        # Mnemonic is too short: 10 words instead of 11
+        partial_mnemonic = "abandon " * 9 + "about"
+        mnemonic_generation.calculate_checksum(partial_mnemonic.split(" "), wordlist_language_code=SettingsConstants.WORDLIST_LANGUAGE__ENGLISH)
+
+    with pytest.raises(Exception) as e:
+        # Valid mnemonic but unsupported length
+        mnemonic = "devote myth base logic dust horse nut collect buddy element eyebrow visit empty dress jungle"
+        mnemonic_generation.calculate_checksum(mnemonic.split(" "), wordlist_language_code=SettingsConstants.WORDLIST_LANGUAGE__ENGLISH)
+
+    with pytest.raises(Exception) as e:
+        # Mnemonic is too short: 22 words instead of 23
+        partial_mnemonic = "abandon " * 21 + "about"
+        mnemonic_generation.calculate_checksum(partial_mnemonic.split(" "), wordlist_language_code=SettingsConstants.WORDLIST_LANGUAGE__ENGLISH)
+
+    with pytest.raises(Exception) as e:
+        # Invalid BIP-39 word
+        partial_mnemonic = "foobar " * 9 + "about"
```

Comment:
I believe this will pass on the branch that checks for word length and raises exception because 10 words, rather than having the exception raised in embit.bip39.mnemonic_to_bytes()... which more precisely will raise ValueError

Reply:
Our brains are in sync! I just noticed that as well!